### PR TITLE
Added Behavioral Analytics APIs

### PR DESCRIPTION
**Pull request by carlosdelest**
Add Behavioral Analytics APIs.

POST is missing - I understood from @afoucret and @joemcelroy that this API should not yet be available for clients. It can be contributed as a separate PR however.

### CODE REVIEW

File: specification/search_application/get_behavioral_analytics/BehavioralAnalyticsGetRequest.ts
Comment (carlosdelest) [inline specification/search_application/get_behavioral_analytics/BehavioralAnalyticsGetRequest.ts:1]: Overall question for @sethmlarson : Why are these APIs not grouped at the REST Spec level on their own namespace, like `behavioral_analytics.get`?

In my mind, they make more sense grouped separately from search applications. WDYT?

Comment (sethmlarson) [reply specification/search_application/get_behavioral_analytics/BehavioralAnalyticsGetRequest.ts:1]: If they aren't a submodel of a Search Application they can have their own top-level namespace. In that case I'd recommend having a different root path segment like `/_analytics` or `/_behavioral_analytics`.

**Review by sethmlarson — APPROVED** [review]
These changes as-is LGTM, will need to be updated if the API gets its namespace changed.

File: specification/search_application/get_behavioral_analytics/BehavioralAnalyticsGetRequest.ts
Comment (carlosdelest) [reply specification/search_application/get_behavioral_analytics/BehavioralAnalyticsGetRequest.ts:1]: Right now we have:
`/_application/search_applications` for search apps, and 
`/_application/analytics` for analytics.

I'd suggest we group them on a different namespace, or that both search apps and analytics have a common `application` namespace.

WDYT?

**Comment by carlosdelest** [comment]
Hey @sethmlarson , I added a PR for changing the namespace [here](https://github.com/elastic/elasticsearch/pull/96225). Can you please double check it?

Once the above is merged, I'll update this PR with the changes.

**Comment by carlosdelest** [comment]
I'll merge this PR as-is, and we can deal with namespace changes later

**Pull request merged**
